app/gui/src/project-view/components/ExtendedMenu.vue

```diff
@@ -1,77 +1,31 @@
-import SvgButton from '@/components/SvgButton.vue'
 import SvgIcon from '@/components/SvgIcon.vue'
-import { ref } from 'vue'
 
-const showCodeEditor = defineModel<boolean>('showCodeEditor', { required: true })
-const showDocumentationEditor = defineModel<boolean>('showDocumentationEditor', { required: true })
-const props = defineProps<{
-  zoomLevel: number
-}>()
-const emit = defineEmits<{
-  zoomIn: []
-  zoomOut: []
-  fitToAllClicked: []
-}>()
-
-const open = ref(false)
-
-const toggleCodeEditorShortcut = codeEditorBindings.bindings.toggle.humanReadable
-const toggleDocumentationEditorShortcut = documentationEditorBindings.bindings.toggle.humanReadable
+const open = defineModel<boolean>('open', { default: false })
 </script>
 
 <template>
   <DropdownMenu
     v-model:open="open"
-    placement="bottom-end"
+    placement="bottom-start"
     class="ExtendedMenu"
     title="Additional Options"
   >
     <template #button><SvgIcon name="3_dot_menu" class="moreIcon" /></template>
     <template #menu>
       <MenuPanel>
-        <div>
-          <div class="nonInteractive"><SvgIcon name="zoom" class="rowIcon" />Zoom</div>
-          <div class="zoomControl rightSide">
-            <SvgButton
-              class="zoomButton"
-              name="minus"
-              title="Decrease Zoom"
-              @click="emit('zoomOut')"
-            />
-            <span
-              class="zoomScaleLabel"
-              v-text="props.zoomLevel ? props.zoomLevel.toFixed(0) + '%' : '?'"
-            ></span>
-            <SvgButton
-              class="zoomButton"
-              name="add"
-              title="Increase Zoom"
-              @click="emit('zoomIn')"
-            />
-            <div class="divider"></div>
-            <SvgButton
-              name="show_all"
-              class="showAllIcon"
-              title="Show All Components"
-              @click="emit('fitToAllClicked')"
-            />
-          </div>
-        </div>
-        <MenuButton v-model="showCodeEditor" @click="open = false">
-          <SvgIcon name="bottom_panel" class="rowIcon" />
-          Code Editor
-          <div class="rightSide" v-text="toggleCodeEditorShortcut" />
-        </MenuButton>
-        <MenuButton v-model="showDocumentationEditor" @click="open = false">
-          <SvgIcon name="right_panel" class="rowIcon" />
-          Documentation Editor
-          <div class="rightSide" v-text="toggleDocumentationEditorShortcut" />
-        </MenuButton>
+        <MenuEntry action="graph.renameProject" @click="open = false" />
+        <MenuEntry action="graph.refreshExecution" />
+        <MenuEntry action="graph.recomputeAll" />
+        <MenuEntry action="graph.undo" />
+        <MenuEntry action="graph.redo" />
+        <MenuEntry action="graph.addComponent" />
+        <MenuEntry action="graph.fitAll" />
+        <MenuEntry action="graph.toggleCodeEditor" />
+        <MenuEntry action="graph.toggleDocumentationEditor" />
```

Comment:
There's `ActionMenu` component doing exactly this: you just pass an array of Actions. See it's usage in ComponentMenu.vue